### PR TITLE
Add footer with note and data credit

### PR DESCRIPTION
**Pull request by daanlouter**
Let me know if you have any other thoughts. Will make a separate repo for this after merge so we can re-use in other templates.

apr-24-2018 15-27-41

### CODE REVIEW

**Comment by danielgavrilov** [comment]
Looks good, but what if there are multiple sources/credits?

(Would be far nicer if there was a way to write and style this directly on the vis than in a "settings panel"...)

**Comment by theduncanclark** [comment]
> +100, but not possible unfortunately :(

Until Michael's Friday project goes live 👎 
)

> what if there are multiple sources/credits?

Good point though, we could support markdown for multiple links. But harder for non-techies to understand.

**Comment by theduncanclark** [comment]
Oh, right, didn't realise it supported HTML. Maybe that's fine then!

**Comment by daanlouter** [comment]
In principe it's fine. But also think that adding 2 sources shouldn't be a "techie" thing to do (ie. writing HTML). I'll just add settings for 2 more sources.

**Comment by daanlouter** [comment]
Now has an option to add 2 more sources

screen shot 2018-04-26 at 17 59 50

**Comment by theduncanclark** [comment]
So much winning!

**Comment by theduncanclark** [comment]
One quick thought so far: URLs always need to open in a new window and always as full urls. Currently if you miss off the https:// you end up at template.flourish.blah/www.whatevs.com, inside the preview iframe :-)

**Comment by daanlouter** [comment]
>Currently if you miss off the https:// you end up at template.flourish.blah/www.whatevs.com

Solved it like this:
```function addHttp(url) {
	if (url.indexOf("http://") !== 0 && url.indexOf("https://") !== 0) return "http://" + url;
	else return url;
}```

**Comment by daanlouter** [comment]
Would be great if either @danielgavrilov or @theduncanclark could quickly glance over the code before merging!

File: src/lib/footer.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	footer_source_url_3: "",
+	footer_note: "",
+	footer_size: "12",
+	footer_color: "#333333",
+	footer_align: "left",
+	footer_margin: "10",
+	footer_margin_advanced: false,
+	footer_margin_top: 10,
+	footer_margin_right: 10,
+	footer_margin_bottom: 0,
```
[the remark this answers is not in the document]
Comment (daanlouter) [reply src/lib/footer.js]: Mainly because there's already space below embedded iframes on most websites, so don't want to add extra space.

**Comment by danielgavrilov** [comment]
I'm not very familiar with this template, but looks fine to merge!

**Pull request merged**